torch/csrc/jit/runtime/symbolic_script.cpp

```diff
@@ -1440,6 +1440,12 @@ const std::vector<std::string> functions = {
                 return None, None
             return torch.ne(self, other), backward
 
+        def hardshrink(self, lambd: number):
+          def backward(grad_output):
+            mask = ((self > float(lambd)) | (self < -float(lambd))).type_as(self)
```

Comment:
I don't think you need the `type_as(self)` call, since you're multiplying it by `grad_output` type promotion should kick in and we get the correct output type.

Reply:
yea, i talked with @ailzhang about this, they're not actually needed. I can remove it in a follow up. They were needed before we had more robust type promotion.